qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/ConvTypes.h

```diff
@@ -0,0 +1,202 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTIDQT_INDIRECT_CONVTYPES_H_
+#define MANTIDQT_INDIRECT_CONVTYPES_H_
+
+#include "DllConfig.h"
+#include "MantidAPI/FunctionFactory.h"
+#include "MantidAPI/IFunction.h"
+#include "MantidKernel/WarningSuppressions.h"
+
+#include <QMap>
+#include <QStringList>
+#include <boost/optional.hpp>
+#include <unordered_map>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+namespace IDA {
+namespace ConvTypes {
+
+using namespace Mantid::API;
+
+enum class FitType {
+  None,
+  OneLorentzian,
+  TwoLorentzians,
+  TeixeiraWater,
+  StretchedExpFT,
+  ElasticDiffSphere,
+  ElasticDiffRotDiscreteCircle,
+  InelasticDiffSphere,
+  InelasticDiffRotDiscreteCircle,
+};
+
+extern std::map<FitType, bool> FitTypeQDepends;
+extern std::unordered_map<FitType, std::string> FitTypeEnumToString;
+extern std::unordered_map<std::string, FitType> FitTypeStringToEnum;
+
+enum class BackgroundType { None, Flat, Linear };
+
+enum class ParamID {
+  NONE,
+  LOR1_AMPLITUDE,
+  LOR1_PEAKCENTRE,
+  LOR1_FWHM,
+  LOR2_AMPLITUDE_1,
+  LOR2_PEAKCENTRE_1,
+  LOR2_FWHM_1,
+  LOR2_AMPLITUDE_2,
+  LOR2_PEAKCENTRE_2,
+  LOR2_FWHM_2,
+  TW_HEIGHT,
+  TW_DIFFCOEFF,
+  TW_TAU,
+  TW_CENTRE,
+  DELTA_HEIGHT,
+  SE_HEIGHT,
+  SE_TAU,
+  SE_BETA,
+  SE_CENTRE,
+  EDP_HEIGHT,
+  EDP_CENTRE,
+  EDP_RADIUS,
+  IDP_INTENSITY,
+  IDP_RADIUS,
+  IDP_DIFFUSION,
+  IDP_SHIFT,
+  IDRDC_INTENSITY,
+  IDRDC_RADIUS,
+  IDRDC_DECAY,
+  IDRDC_SHIFT,
+  EDRDC_HEIGHT,
+  EDRDC_CENTRE,
+  EDRDC_RADIUS,
+  FLAT_BG_A0,
+  LINEAR_BG_A0,
+  LINEAR_BG_A1,
+};
+
+QString paramName(ParamID id);
+
+inline ParamID &operator++(ParamID &id) {
```

Comment:
Can we drop these inline - the compiler is free to follow or ignore these and it's still subject to the inliner's metrics

https://stackoverflow.com/a/17
https://devblogs.microsoft.com/cppblog/inlining-decisions-in-visual-studio/59575

Reply:
These inlines are needed for the linker so not complain about multiple definitions of these methods. Using the meaning of inline below.

inline - this function will be defined in multiple translation units, don't worry about it. The linker needs to make sure all translation units use a single instance of the variable/function.